OpenSSL/crypto.py

```diff
@@ -2011,27 +2065,123 @@ def add_revoked(self, revoked):
             _raise_current_error()
 
 
-    def export(self, cert, key, type=FILETYPE_PEM, days=100,
-               digest=_UNSPECIFIED):
+    def get_issuer(self):
         """
-        Export a CRL as a string.
+        Get the CRL's issuer.
 
-        :param cert: The certificate used to sign the CRL.
-        :type cert: :py:class:`X509`
+        .. versionadded:: 0.16
 
-        :param key: The key used to sign the CRL.
-        :type key: :py:class:`PKey`
+        :return: :py:class:`X509Name`
+        """
+        _issuer = _lib.X509_NAME_dup(_lib.X509_CRL_get_issuer(self._crl))
+        if _issuer == _ffi.NULL:
+            _raise_current_error()
+        _issuer = _ffi.gc(_issuer, _lib.X509_NAME_free)
+        issuer = X509Name.__new__(X509Name)
+        issuer._name = _issuer
+        return issuer
 
-        :param type: The export format, either :py:data:`FILETYPE_PEM`,
-            :py:data:`FILETYPE_ASN1`, or :py:data:`FILETYPE_TEXT`.
 
-        :param int days: The number of days until the next update of this CRL.
+    def set_version(self, version):
+        """
+        Set the CRL version.
+
+        .. versionadded:: 0.16
+
+        :param int version: The version of the CRL.
+        :return: :py:const:`None`
+        """
+        if _lib.X509_CRL_set_version(self._crl, version) == 0:
+            _raise_current_error()
+
+
+    def _set_boundary_time(self, which, when):
+        return _set_asn1_time(which(self._crl), when)
+
+
+    def set_lastUpdate(self, when):
+        """
+        Set when the CRL was last updated.
+
+        The timestamp is formatted as an ASN.1 GENERALIZEDTIME::
+
+            YYYYMMDDhhmmssZ
+            YYYYMMDDhhmmss+hhmm
+            YYYYMMDDhhmmss-hhmm
+
+        .. versionadded:: 0.16
+
+        :param bytes when: A timestamp string.
+        :return: :py:const:`None`
+        """
+        return self._set_boundary_time(_lib.X509_CRL_get_lastUpdate, when)
+
+
+    def set_nextUpdate(self, when):
```

Comment:
The only "good" reason I have is that other API in this module use this format. I'd insist that we keep it consistent, but I would also be happy to change the other API to use a datetime if the maintainers are okay with a backwards incompatible API change.